packages/cdk/lib/stacks/nested/batch-stack.ts

```diff
@@ -15,35 +16,44 @@ export interface BatchStackProps extends NestedStackProps {
    * Parameters determined by the context.
    */
   readonly contextParameters: ContextAppParameters;
+  /**
+   * Request Spot capacity to be created
+   */
+  readonly createSpotBatch: boolean;
+  /**
+   * Request On-Demand capacity to be created
+   */
+  readonly createOnDemandBatch: boolean;
 }
 
 export class BatchStack extends NestedStack {
```

Comment:
Thoughts on changing this class to render batches through a public method rather than through the constructor? It would simplify a lot of the logic both here and in the context stack above.

```
export class BatchStack extends NestedStack {
  public readonly vpc: IVpc;
  public readonly artifactBucket: IBucket;
  public readonly outputBucket: IBucket;
  public readonly readBucketArns: string[];
  public readonly readWriteBucketArns: string[];
  public readonly instanceTypes?: InstanceType[];

  constructor(scope: Construct, id: string, props: BatchStackProps) {
    super(scope, id, props);

    const { vpc, contextParameters } = props;
    this.vpc = vpc;
    this.artifactBucket = BucketOperations.importBucket(this, "ArtifactBucket", contextParameters.artifactBucketName);
    this.outputBucket = BucketOperations.importBucket(this, "OutputBucket", contextParameters.outputBucketName);

    this.readBucketArns = contextParameters.readBucketArns ?? [];
    this.readWriteBucketArns = contextParameters.readWriteBucketArns ?? [];

    this.instanceTypes = contextParameters.instanceTypes;
  }

  public renderBatch(id: string, computeType?: ComputeType): Batch {
    const batch = new Batch(this, id, {
      vpc: this.vpc,
      instanceTypes: this.instanceTypes,
      computeType,
      launchTemplateData: LAUNCH_TEMPLATE,
      awsPolicyNames: ["AmazonSSMManagedInstanceCore", "CloudWatchAgentServerPolicy"],
      resourceTags: this.nestedStackParent?.tags.tagValues(),
    });

    BucketOperations.grantBucketAccess(this, batch.role, (this.readBucketArns).concat(this.artifactBucket.bucketArn), true);
    BucketOperations.grantBucketAccess(this, batch.role, (this.readWriteBucketArns).concat(this.outputBucket.bucketArn));

    this.artifactBucket.grantRead(batch.role);
    this.outputBucket.grantReadWrite(batch.role);

    return batch;
  }
}
```

Reply:
Thanks, i think this is a good idea to not plumb the state through two different stacks. More I look at it less value of a nested stack for batch I see here all together. I'll ping you on that.

Reply:
As we have agreed on this class would remain `constructor()` based and consuming it's configuration from props, instead of allowing outside code to create resources within this stack instead by invoking `public renderBatch()` method potentially unbound number of times.

The decision comes from us wanting to keep the nested stacks separated for the time being, the other option discussed has been abolish the nested stack for batch altogether and just render required compute environments where we have our engines rendered.